src/test/java/edu/unc/lib/boxc/migration/cdm/services/MigrationProjectFactoryTest.java

```diff
@@ -0,0 +1,235 @@
+            fail();
+        } catch (InvalidProjectStateException e) {
+            assertTrue(e.getMessage().contains("directory already contains a migration project"));
+        }
+
+        // Original project should still exist and be unchanged
+        assertEquals(projectsBase.resolve(PROJ_NAME), project.getProjectPath());
+
+        assertReturnedPropertiesPopulated(project, PROJ_NAME, PROJ_NAME);
+        assertPropertiesFilePopulated(project, PROJ_NAME, PROJ_NAME);
+    }
+
+    @Test
+    public void loadPathNotExistTest() throws Exception {
+        try {
+            MigrationProjectFactory.loadMigrationProject(projectsBase.resolve(PROJ_NAME));
+            fail();
+        } catch (InvalidProjectStateException e) {
+            assertTrue(e.getMessage().contains("does not exist"));
+        }
+        assertTrue(Files.notExists(projectsBase.resolve(PROJ_NAME)));
+    }
+
+    @Test
+    public void loadNotAProjectTest() throws Exception {
+        Path projectPath = projectsBase.resolve(PROJ_NAME);
+        Files.createDirectory(projectPath);
+        try {
+            MigrationProjectFactory.loadMigrationProject(projectPath);
+            fail();
+        } catch (InvalidProjectStateException e) {
+            assertTrue(e.getMessage().contains("does not contain an initialized project"));
+        }
+        assertTrue(Files.exists(projectPath));
+        try(DirectoryStream<Path> dirStream = Files.newDirectoryStream(projectPath)) {
+            assertFalse("Existing directory should remain empty", dirStream.iterator().hasNext());
+        }
+    }
+
+    @Test
+    public void loadValidProject() throws Exception {
+        Path projectPath = projectsBase.resolve(PROJ_NAME);
+        MigrationProject projectCreated = MigrationProjectFactory
+                .createMigrationProject(projectsBase, PROJ_NAME, COLL_ID, USERNAME);
+
+        MigrationProject projectLoaded = MigrationProjectFactory.loadMigrationProject(projectPath);
+
+        assertNotNull(projectLoaded);
+        assertEquals(projectPath, projectLoaded.getProjectPath());
+
+        assertReturnedPropertiesPopulated(projectLoaded, PROJ_NAME, COLL_ID);
+        assertPropertiesFilePopulated(projectLoaded, PROJ_NAME, COLL_ID);
+
+        assertEquals("Expect created and loaded projects to have same timestamp",
+                projectCreated.getProjectProperties().getCreatedDate(),
+                projectLoaded.getProjectProperties().getCreatedDate());
+    }
+
+    private void assertPropertiesFilePopulated(MigrationProject project, String expName, String expCollId)
+            throws IOException {
+        Path propertiesPath = project.getProjectPropertiesPath();
+        assertTrue("Properties object must exist", Files.exists(propertiesPath));
+        MigrationProjectProperties properties = ProjectPropertiesSerialization.read(propertiesPath);
+        assertPropertiesSet(properties, expName, expCollId);
```

Comment:
I suppose it doesn't matter too much, but should this be `assertReturnedPropertiesPopulate`?

Reply:
No, `assertReturnedPropertiesPopulated` checks the properties directly held by a MigrationProject instance, while `assertPropertiesFilePopulated` checks the properties held in the project.json file persisted on disk. `assertPropertiesSet` checks the values of properties regardless of the source, which is why its used by both of the other methods.